apps/files_sharing/src/components/SharingEntryLink.vue

```diff
@@ -407,6 +408,29 @@ export default {
 			return this.config.isDefaultExpireDateEnforced && this.share && !this.share.id
 		},
 
+		sharePolicyHasRequiredProperties() {
+			return this.config.enforcePasswordForPublicLink || this.config.isDefaultExpireDateEnforced
+		},
+
+		requiredPropertiesMissing() {
+			// Destructure share and config from this
+			const { share, config } = this
+			// Ensure share exist and the share policy has required properties
+			if (!this.sharePolicyHasRequiredProperties) return false
+
+			if (share) {
+				// If share has ID, then this is an incoming link share created from the existing link share
+				// Hence assume required properties
+				if (share.id) return true
```

Comment:
```suggestion
				if (share.id) {
					return true
				}
```

We never use short syntax.